### PR TITLE
Remove deprecated `mode` argument from ModelSummary

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Part of #10312 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/model_summary.py [hunk 1]
```diff
@@ -436,40 +415,17 @@ def _is_lazy_weight_tensor(p: Tensor) -> bool:
     return False
 
 
-def summarize(
-    lightning_module: "pl.LightningModule", mode: Optional[str] = None, max_depth: Optional[int] = None
-) -> ModelSummary:
+def summarize(lightning_module: "pl.LightningModule", max_depth: Optional[int] = None) -> ModelSummary:
```
Comment (ananthsub) [inline pytorch_lightning/utilities/model_summary.py:418]: max_depth shouldn't be Optional anymore, and the default value should be 1.

File: pytorch_lightning/core/lightning.py [hunk 2]
```diff
@@ -1689,19 +1689,14 @@ def tbptt_split_batch(self, batch, split_size):
 
         return splits
 
-    def summarize(self, mode: Optional[str] = "top", max_depth: Optional[int] = None) -> Optional[ModelSummary]:
+    def summarize(self, max_depth: Optional[int] = None) -> Optional[ModelSummary]:
```
Comment (ananthsub) [inline pytorch_lightning/core/lightning.py:1692]: Same here

File: pytorch_lightning/utilities/model_summary.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/utilities/model_summary.py]: Updated!

**Comment by carmocca** [comment]
Hey @kaushikb11, I had https://github.com/PyTorchLightning/pytorch-lightning/pull/10446 already opened. Do you mind merging that and then fully removing the argument here?

File: tests/deprecated_api/test_remove_1-6.py [hunk 3]
```diff
@@ -77,13 +76,11 @@ def test_v1_6_0_train_loop(tmpdir):
...
-    with pytest.deprecated_call(match="Argument `mode` in `ModelSummary` is deprecated in v1.4"):
-        ModelSummary(model, mode="top")
-
-    with pytest.deprecated_call(match="Argument `mode` in `LightningModule.summarize` is deprecated in v1.4"):
-        model.summarize(mode="top")
+def test_v1_6_0_rank_zero_warnings_moved():
+    with pytest.deprecated_call(match="in v1.3.7 and will be removed in v1.6"):
+        rank_zero_warn("test")
+    with pytest.deprecated_call(match="in v1.3.7 and will be removed in v1.6"):
+        rank_zero_deprecation("test")
```
Comment (kaushikb11) [inline tests/deprecated_api/test_remove_1-6.py:83]:
```suggestion
```

**Review by rohitgr7 — APPROVED** [review]

File: tests/utilities/test_model_summary.py [hunk 4]
```diff
@@ -141,19 +141,12 @@ def forward(self, inp):
 
 def test_invalid_weights_summmary():
     """Test that invalid value for weights_summary raises an error."""
```
Comment (kaushikb11) [inline tests/utilities/test_model_summary.py:143]:
```suggestion
    """Test that invalid value for weights_summary raises an error."""
    model = LightningModule()
```

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Pull request merged**